test/client-reconnect.js

```diff
@@ -6,36 +6,42 @@ const { createServer } = require('http')
 const FakeTimers = require('@sinonjs/fake-timers')
 
```

Comment:
I am not sure I got this right and completely understood how the reconnection works in this case.

Reply:
I have seen similar tests in `client-request.js` if I am not mistaken.